payload/reggae/rules/cmake.d

```diff
@@ -161,14 +174,26 @@ private bool isHeaderFile(in string file) {
     return isCHeader(file) || isCppHeader(file);
 }
 
-private string getLinkerFlags(in imported!"std.json".JSONValue target) {
+private struct CMakeLinkerFlags {
+    import reggae.types : Flags, LibraryFlags;
+    Flags flags;
+    LibraryFlags libraryFlags;
+    Flags libraryPathFlags;
+
+    Flags getRegularFlags() const {
+        return Flags(flags.value.dup ~ libraryPathFlags.value.dup);
```

Comment:
Do they need to be duplicated?

Reply:
I was trying to avoid the 

> Error: `inout` constructor `reggae.types.Flags.this` creates const object, not mutable

Deleted this method and moved the work to the caller function.